common_types/src/record_batch.rs

```diff
@@ -386,22 +390,18 @@ impl RecordBatchWithKey {
         }
     }
 
-    /// Select the rows according to the `selected_rows`.
-    ///
-    /// The data retains intact if failed.
-    pub fn select_data(&mut self, selected_rows: &[bool]) -> Result<()> {
-        assert_eq!(self.num_rows(), selected_rows.len());
-
+    /// Select the rows according to the `filter_array`.
+    pub fn select_data(&mut self, filter_array: &BooleanArray) -> Result<()> {
+        assert_eq!(self.num_rows(), filter_array.len());
         let selected_record_batch =
-            operation::select_record_batch(&self.data.arrow_record_batch, selected_rows)
+            filter_record_batch(&self.data.arrow_record_batch, filter_array)
```

Comment:
For free functions, we may use it with its super module as prefix:
```suggestion
            compute::filter_record_batch(&self.data.arrow_record_batch, filter_array)
```